tests/phpunit/tests/block-supports/block-style-variations.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/**
+ * Test the block style variations block support.
+ *
+ * @package WordPress
+ * @subpackage Block Supports
+ * @since 6.6.0
+ *
+ * @group block-supports
+ */
+class WP_Block_Supports_Block_Style_Variations_Test extends WP_UnitTestCase {
+	/**
+	 * Theme root directory.
+	 *
+	 * @var string|null
+	 */
+	private $theme_root;
+
+	/**
+	 * Original theme directory.
+	 *
+	 * @var array|null
+	 */
+	private $orig_theme_dir;
+
+	public function set_up() {
+		parent::set_up();
+		$this->theme_root = realpath( DIR_TESTDATA . '/themedir1' );
+
+		$this->orig_theme_dir = $GLOBALS['wp_theme_directories'];
+
+		// /themes is necessary as theme.php functions assume /themes is the root if there is only one root.
+		$GLOBALS['wp_theme_directories'] = array( WP_CONTENT_DIR . '/themes', $this->theme_root );
+
+		add_filter( 'theme_root', array( $this, 'filter_set_theme_root' ) );
+		add_filter( 'stylesheet_root', array( $this, 'filter_set_theme_root' ) );
+		add_filter( 'template_root', array( $this, 'filter_set_theme_root' ) );
+
+		// Clear caches.
+		wp_clean_themes_cache();
+		unset( $GLOBALS['wp_themes'] );
+	}
+
+	public function tear_down() {
+		$GLOBALS['wp_theme_directories'] = $this->orig_theme_dir;
+		wp_clean_themes_cache();
+		unset( $GLOBALS['wp_themes'] );
+
+		// Reset data between tests.
+		wp_clean_theme_json_cache();
+		parent::tear_down();
+	}
+
+	public function filter_set_theme_root() {
+		return $this->theme_root;
+	}
+
+	/**
+	 * Tests that block style variations registered via either
+	 * `register_block_style` with a style object, or a standalone block style
+	 * variation file within `/styles`, are added to the theme data.
+	 *
+	 * @ticket 61312
+	 */
+	public function test_add_registered_block_styles_to_theme_data() {
+		switch_theme( 'block-theme' );
+
+		$variation_styles_data = array(
+			'color'    => array(
+				'background' => 'darkslateblue',
+				'text'       => 'lavender',
+			),
+			'blocks'   => array(
+				'core/heading' => array(
+					'color' => array(
+						'text' => 'violet',
+					),
+				),
+			),
+			'elements' => array(
+				'link' => array(
+					'color'  => array(
+						'text' => 'fuchsia',
+					),
+					':hover' => array(
+						'color' => array(
+							'text' => 'deeppink',
+						),
+					),
+				),
+			),
+		);
+
+		register_block_style(
+			'core/group',
+			array(
+				'name'       => 'my-variation',
+				'style_data' => $variation_styles_data,
+			)
+		);
+
+		$theme_json   = WP_Theme_JSON_Resolver::get_theme_data()->get_raw_data();
+		$group_styles = $theme_json['styles']['blocks']['core/group'] ?? array();
```

Comment:
Just checking `??` is okay in tests 

I think it's still not "officially" approved for use in  Core source files, but tests might be fine.

Reply:
I just looked and `??` is used in several files already in core. My understanding is this is ok given the minimum PHP requirements for WordPress now.

Reply:
Some of the places that operator is already in use include:

- Interactivity API
- Elements block support
- Dimensions block support
- Plugin install
- Translation controller
- Style engine
- Blocks including Navigation, Image, Calendar, Avatar, Featured Image, Gallery
- A range of phpunit tests

I'm going to mark these operator comments as resolved but happy to address them if all those examples aren't kosher 🙂

Reply:
This is the way. Thanks for pulling up those examples 👍🏻 

For the record, here's an open ticket that discusses about discussing the use of `??` in Core.

https://core.trac.wordpress.org/ticket/58874

Reply:
Seems silly not to use `??`. It's part of the language.

Reply:
I think it's generally avoided as it can be confusing. The use of [short ternaries](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#ternary-operator) is explicitly ruled out and given null coalescing is very similar I think it's best to avoid it.

It's also an absolute bugger to google if you're trying to remember which syntax does what.

Reply:
I don't believe it's avoided for being confusing, and I'd hesitate to put it in the short ternary bucket 😅 as it's a very useful operator, allowing us to replace `isset`. Personally I think it makes the code more readable. 

The main reason it's not in the standards is nobody has added it in there yet, as per the [trac ticket discussion](https://core.trac.wordpress.org/ticket/58874). It's sad that we should be blocked from using a helpful language feature due to bureaucratic detail, but there we are 🤷